### PR TITLE
Add missing product handle API.

### LINKED ISSUE #194: Add missing API: get product by handle
Add support for retrieving a product by its handle. Necessary for following share links when the object does not exist.

### PR DESCRIPTION
**Pull request by dbart01**
### What this does
- adds missing API that allows fetching a product by its unique `handle`
- adds integration test for product handle API

Fixes #194 
Relies on #204

@bgulanowski @davidmuzi

### CODE REVIEW

**Comment by dbart01** [comment]
Ready for review.

File: Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m [hunk 1]
```diff
@@ -75,19 +75,31 @@ - (BUYRequestOperation *)getProductsPage:(NSUInteger)page completion:(BUYDataPro
...
 	return [self getProductsByIds:@[productId] completion:^(NSArray *products, NSError *error) {
-		if (products.count > 0) {
-			block(products[0], error);
-		} else {
-			if (!error) {
-				error = [NSError errorWithDomain:BUYShopifyErrorDomain code:BUYShopifyError_InvalidProductID userInfo:@{ NSLocalizedDescriptionKey : @"Product ID is not valid. Confirm the product ID on your shop's admin and also ensure that the visibility is on for the Mobile App channel." }];
-			}
-			block(nil, error);
-		}
+		block(products.firstObject, error);
```
Comment (davidmuzi) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: Intend to use this?
`productsError`

Comment (dbart01) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: It's used internally by `getProductsByIds:completion:` so the `error` in this case will never be `nil`.

File: Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m [hunk 2]
```diff
@@ -75,19 +75,31 @@ - (BUYRequestOperation *)getProductsPage:(NSUInteger)page completion:(BUYDataPro
...
+	
+	NSURL *url = [self urlForProductListingsWithParameters:@{ @"handle" : handle }];
+	return [self getRequestForURL:url completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		
+		BUYProduct *product = nil;
+		if (json && !error) {
+			product = [self.modelManager insertProductsWithJSONArray:json[BUYProductsKey]].firstObject;
+		}
+		
+		if (!product && !error) {
```
Comment (bgulanowski) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: Seems like we should really get an error from the endpoint if the handle is not recognized.

Comment (dbart01) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: If the `handle` provided isn't found, the server will return an empty collection. In this case, that means the `product` and `error` will be `nil`.

Comment (bgulanowski) [inline Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m]: 👍

**Comment by bgulanowski** [comment]
LGTM.

**Pull request merged**